src/consumers.ts

```diff
@@ -187,8 +200,88 @@ Recommendation: ${recommendedVersion}`;
                 source: '\nDependency Analytics Plugin [Powered by Snyk]',
             };
 
+            let diagnosticLinePresent = false
+            let diagnosticIndex = 0
+            for (let i = 0; i < cls.diagnostics.length; i++) {
+                if (cls.diagnostics[i].range.start.line.toString() == diagnostic.range.start.line.toString()) {
+                    diagnosticLinePresent = true
+                    diagnosticIndex = i
+                }
+            }
+
+            if (diagnosticLinePresent) {
+                // Add new counts to old one
+                const oldMessage = cls.diagnostics[diagnosticIndex].message.split("\n")
+                const packageCount = parseInt(oldMessage[1].split(":")[1].trim()) + 1
+                const securityVulnerabilityCount = parseInt(oldMessage[2].split(":")[1].trim()) + this.vulnerabilityCount
+                const securityAdvisoryCount = parseInt(oldMessage[3].split(":")[1].trim()) + this.advisoryCount
+
+                // Compute sum of exploits, taking 'unavailable' value for exploit.
+                let oldExploit = oldMessage[4].split(":")[1].trim()
+                var newExploitCount: any
+                if (oldExploit == "unavailable") {
+                    newExploitCount = this.exploitCount || "unavailable";
+                } else {
+                    newExploitCount = parseInt(oldExploit) + this.exploitCount
+                }
+
+                // Compute highest severity
+                const oldSeverity = oldMessage[5].split(":")[1].trim()
+                var newSeverity = oldSeverity
+                if (this.highestSeverity == "critical" || oldSeverity == "critical")
+                    newSeverity = "critical"
+                else if (this.highestSeverity == "high" || oldSeverity == "high")
+                    newSeverity = "high"
+                else if (this.highestSeverity == "medium" || oldSeverity == "medium")
+                    newSeverity = "medium"
+                else if (this.highestSeverity == "low" || oldSeverity == "low")
+                    newSeverity = "low"
+                
+                // Compute rec version based on max criteria
+                const oldRecVersion = oldMessage[6].split(":")[1].trim()
+                var newRecVersion = oldRecVersion
+                if (oldRecVersion == "N/A") {
+                    newRecVersion = this.changeTo || "N/A";
```

Comment:
It is bit hard to understand with too many statements in one go, Can you split that into multiple functions?(Extract method).

Reply:
New code I have split all these into smaller functions.